### DISCUSSION TITLE
Unicode characters don't appear to work

### DISCUSSION THREAD

**Discussion by ex-howiebbq** [allejo/jekyll-toc#47] [Q&A] [answered]
Hello, 

I have two issues
I have used your plugin on my website. 

First, I use the Vietnamese language to write heading on pages (picture 1) for creating ToC. However, I meet issues like picture 2 and when I click ToC, it doesn't scroll. 
Second, I viewed on page source, it showed like <a href="#theoretical-v%C3%A0-empirical-l%C3%BD-thuy%E1%BA%BFt-v%C3%A0-th%E1%BB%B1c-nghi%E1%BB%87m" . I think this link a bad problem for SEO.

I don't know how to fix it. Could you give me some solution? 

![toc](https://user-images.githubusercontent.com/73570691/97783284-48483400-1bc9-11eb-8bf4-b89797623905.png)
![toc_1](https://user-images.githubusercontent.com/73570691/97783285-4a11f780-1bc9-11eb-9656-4e425adbc7f9.png)

**Comment by allejo** [comment]
Do you have an example repository I can look at? I'm currently using kramdown and UTF-8 characters are loading fine for me.

```twig
{% capture markdown %}
# Theoretical và-empirical-lý-thuyết-và-thực-nghiệm
{% endcapture %}
{% assign text = markdown | markdownify %}

{% include toc.html html=text class="awesome-toc-class" %}
```

```html
<ul class="awesome-toc-class">
  <li><a href="#theoretical-và-empirical-lý-thuyết-và-thực-nghiệm">Theoretical và-empirical-lý-thuyết-và-thực-nghiệm</a></li>
</ul>
```

**Comment by ex-howiebbq** [comment]
hello, 

Thank your response. I have input files. Please check it

I also using kramdown and UTF-8 characters. 

The first time, I write some of heading included Vietnamese (VN) and English (EN) languages. After that, I clicked TOC, it doesn't scroll to VN heading, only EN. Then I try to copy the link address of heading, it looks like `<a href="#1-theo-ph%C6%B0%C6%A1ng-ph%C3%A1p-x%C3%A1c-%C4%91%E1%BB%8Bnh-chung" class="nomunge">.` 

Second time, I combined VN and EN together. For example,

`<a href="#theoretical-v%C3%A0-empirical-l%C3%BD-thuy%E1%BA%BFt-v%C3%A0-th%E1%BB%B1c-nghi%E1%BB%87m">`

However, I edited content from VN to EN, it runs okay. 

I was put toc_pure_liquid.html on _includes/

[toc_pure_liquid.txt](https://github.com/allejo/jekyll-toc/files/5479727/toc_pure_liquid.txt)

and docs.html on _layouts/

[docs.txt](https://github.com/allejo/jekyll-toc/files/5479776/docs.txt)

**Comment by allejo** [comment]
Oh man, `toc_pure_liquid.txt` was the first version that I published years ago. Can you try using the [latest version](https://github.com/allejo/jekyll-toc/blob/master/_includes/toc.html) and see if the problem persists?

**Comment by ex-howiebbq** [comment]
> Oh man, `toc_pure_liquid.txt` was the first version that I published years ago. Can you try using the [latest version](https://github.com/allejo/jekyll-toc/blob/master/_includes/toc.html) and see if the problem persists?

I update the latest version. However, the issue still shows.  
- ToC doesn't scroll if combined VN and EN characters on sentence or just VN characters
- `<a href="#">` error characters

**Comment by allejo** [comment]
Could you please provide some sample content? When I tried decoding the HTML encoded values and used it for [my tests](https://github.com/allejo/jekyll-toc/issues/47#issuecomment-720041957), I couldn't replicate your behavior.

**Comment by ex-howiebbq** [comment]
Hello,

My blog fork from docs.docker.com, keep all js, css, and their gem. Just put katex js

I check 2 times
- Firstly, I checked the anchor link of heading. it quite good.

![good-anchor-link](https://user-images.githubusercontent.com/73570691/98120722-e5b0aa00-1ee0-11eb-8aac-a4d0033071fc.png)

- Secondly, I checked the ToC list, it showing some Latin characters combine VN characters.

![error](https://user-images.githubusercontent.com/73570691/98120871-198bcf80-1ee1-11eb-8154-7b19a922ef3b.png)

You can go to my link [here](https://statswork.wiki/phuong-phap-nghien-cuu/quantitative-research/bien-so-trong-nghien-cuu/).

**Comment by allejo** [comment, marked as answer]
This doesn't seem to be a problem with the code jekyll-toc is generating. When I disable JS on your website, the TOC links work as intended. This appears to be some JS interfering with the behavior of navigation.

![2020-11-06 09-38-12 2020-11-06 09_39_40](https://user-images.githubusercontent.com/1246453/98397308-04638d80-2014-11eb-9d11-06ca2bce7bb3.gif)
